k8s/pkg/registryserver/registry_cache.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/networkservicemesh/networkservicemesh/k8s/pkg/registryserver/resource_cache"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+type RegistryCache interface {
+	AddNetworkService(ns *v1.NetworkService) (*v1.NetworkService, error)
+	GetNetworkService(name string) (*v1.NetworkService, error)
+
+	AddNetworkServiceManager(nsm *v1.NetworkServiceManager) (*v1.NetworkServiceManager, error)
+	GetNetworkServiceManager(name string) (*v1.NetworkServiceManager, error)
+
+	AddNetworkServiceEndpoint(nse *v1.NetworkServiceEndpoint) (*v1.NetworkServiceEndpoint, error)
+	DeleteNetworkServiceEndpoint(endpointName string) error
+	GetNetworkServiceEndpoints(networkServiceName string) []*v1.NetworkServiceEndpoint
+
+	Start() error
+	Stop()
+}
+
+type registryCacheImpl struct {
+	networkServiceCache         *resource_cache.NetworkServiceCache
+	networkServiceEndpointCache *resource_cache.NetworkServiceEndpointCache
+	networkServiceManagerCache  *resource_cache.NetworkServiceManagerCache
+	clientset                   *nsmClientset.Clientset
+	stopFuncs                   []func()
+}
+
+func NewRegistryCache(clientset *nsmClientset.Clientset) RegistryCache {
+	return &registryCacheImpl{
+		networkServiceCache:         resource_cache.NewNetworkServiceCache(),
+		networkServiceEndpointCache: resource_cache.NewNetworkServiceEndpointCache(),
+		networkServiceManagerCache:  resource_cache.NewNetworkServiceManagerCache(),
+		clientset:                   clientset,
+		stopFuncs:                   make([]func(), 0, 3),
+	}
+}
+
+func (rc *registryCacheImpl) Start() error {
+	factory := externalversions.NewSharedInformerFactory(rc.clientset, 100*time.Millisecond)
+
+	if stopFunc, err := rc.networkServiceCache.Start(factory); err != nil {
+		rc.Stop()
+		return err
+	} else {
+		rc.stopFuncs = append(rc.stopFuncs, stopFunc)
+	}
+
+	if stopFunc, err := rc.networkServiceEndpointCache.Start(factory); err != nil {
+		rc.Stop()
+		return err
+	} else {
+		rc.stopFuncs = append(rc.stopFuncs, stopFunc)
+	}
+
+	if stopFunc, err := rc.networkServiceManagerCache.Start(factory); err != nil {
+		rc.Stop()
+		return err
+	} else {
+		rc.stopFuncs = append(rc.stopFuncs, stopFunc)
+	}
+
+	return nil
+}
+
+func (rc *registryCacheImpl) AddNetworkService(ns *v1.NetworkService) (*v1.NetworkService, error) {
+	nsResponse, err := rc.clientset.NetworkservicemeshV1().NetworkServices("default").Create(ns)
+	if nsResponse != nil {
+		rc.networkServiceCache.Add(nsResponse)
+	}
+	return nsResponse, err
+}
+
+func (rc *registryCacheImpl) GetNetworkService(name string) (*v1.NetworkService, error) {
+	if ns := rc.networkServiceCache.Get(name); ns == nil {
+		return nil, fmt.Errorf("no NetworkService with name: %v", name)
+	} else {
+		return ns, nil
+	}
+}
+
+func (rc *registryCacheImpl) AddNetworkServiceEndpoint(nse *v1.NetworkServiceEndpoint) (*v1.NetworkServiceEndpoint, error) {
+	nseResponse, err := rc.clientset.NetworkservicemeshV1().NetworkServiceEndpoints("default").Create(nse)
```

Comment:
Are we hardcoding the `default` namespace here?
I know that this is all over the code, but we might want to change this.
@edwarnicke what do you think?